benchmark/utils/stencil_matrix.hpp

```diff
@@ -266,15 +266,17 @@ gko::matrix_data<ValueType, IndexType> generate_3d_stencil_box(
     };
     const auto diag_value = static_cast<ValueType>(nnz_in_row() - 1);
 
-    for (IndexType i = 0; i < discretization_points; ++i) {
-        for (IndexType j = 0; j < discretization_points; ++j) {
-            for (IndexType k = 0; k < discretization_points; ++k) {
-                auto row = flat_idx(i, j, k);
-                for (IndexType d_i : {-1, 0, 1}) {
-                    for (IndexType d_j : {-1, 0, 1}) {
-                        for (IndexType d_k : {-1, 0, 1}) {
-                            if (is_valid_neighbor(d_i, d_j, d_k)) {
-                                auto col = flat_idx(i + d_i, j + d_j, k + d_k);
+    A_data.nonzeros.reserve(nnz_in_row() * local_size);
+
+    for (IndexType iz = 0; iz < discretization_points; ++iz) {
+        for (IndexType iy = 0; iy < discretization_points; ++iy) {
+            for (IndexType ix = 0; ix < discretization_points; ++ix) {
+                auto row = flat_idx(iz, iy, ix);
+                for (IndexType dz : {-1, 0, 1}) {
+                    for (IndexType dy : {-1, 0, 1}) {
+                        for (IndexType dx : {-1, 0, 1}) {
+                            if (is_valid_neighbor(dz, dy, dx)) {
+                                auto col = flat_idx(iz + dz, iy + dy, ix + dx);
                                 if (is_in_box(col, static_cast<IndexType>(
                                                        global_size))) {
```

Comment:
A_data contains the global index so the A_data `IndexType` uses the `GlobalIndexType`, right?

Reply:
Yes. Since the stencil generation can also be used for plain CSR matrices, `IndexType` may also be just `int32`. But in the distributed case your statement is true.